src/modules/core/components/checkbox/checkbox.module.scss

```diff
@@ -0,0 +1,118 @@
+@use '@/styles/variables.module';
+
+.container {
+  display: block;
+  position: relative;
+  cursor: pointer;
+  font-size: 22px;
+  user-select: none;
+  padding: 9px;
+  width: fit-content;
+  border-radius: 30%;
+  transition-duration: variables.$transition-fast;
+  transition-property: background-color;
+  &:hover {
+    background-color: rgba(25, 118, 210, 0.04);
+  }
+}
+
+.input {
+  position: absolute;
+  opacity: 0;
+  cursor: pointer;
+  height: 0;
+  width: 0;
+}
+
+.checkmark {
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  top: 0;
+  left: 0;
+  height: 24px;
+  width: 24px;
+  background-color: #fff;
+  border: 1px solid variables.$primary-color;
+  border-radius: 4px;
+  transition: variables.$transition-ultrafast;
+
+  &::before {
+    position: absolute;
+    top: 11px;
+    left: 4px;
+    display: block;
+    content: '';
+    width: 2px;
+    height: 0;
+    background-color: #fff;
+    border-radius: 1px;
+    transform: rotate(-45deg);
+    transform-origin: top right;
+  }
+
+  &::after {
+    position: absolute;
+    top: 16px;
+    left: 7px;
+    display: block;
+    content: '';
+    width: 2px;
+    height: 0;
+    background-color: #fff;
+    border-radius: 1px;
+    transform: rotate(-139deg);
+    transform-origin: top right;
+  }
+
+  &_active {
+    transition: variables.$transition-fast;
+    border-color: transparent;
+    &::before {
+      animation-name: check1;
+      animation-duration: 0.3s;
+      animation-fill-mode: forwards;
+    }
+
+    &::after {
+      animation-name: check2;
+      animation-duration: 0.3s;
+      animation-fill-mode: forwards;
+      animation-delay: 0.15s;
+    }
+  }
+}
+
+label.container input:disabled + span {
```

Comment:
you don't need `label` here. Just remove it. And replase `span` with `.checkmark`